src/OrchardCore/OrchardCore.Testing/Infrastructure/OrchardCoreTestFixture.cs

```diff
@@ -0,0 +1,29 @@
+using System;
+using System.IO;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.AspNetCore.Mvc.Testing;
+using Microsoft.AspNetCore.TestHost;
+using Microsoft.Extensions.Hosting;
+
+namespace OrchardCore.Testing.Infrastructure;
+
+public class OrchardCoreTestFixture<TStartup> : WebApplicationFactory<TStartup> where TStartup : class
+{
+    protected override void ConfigureWebHost(IWebHostBuilder builder)
```

Comment:
override the CreateHost method to configure the IHostBuilder based on the SiteContextOptions.ConfigureHostBuilderAction
eg.
```code
    protected override IHost CreateHost(IHostBuilder builder)
    {
        if (SiteContextOptions.ConfigureHostBuilderAction != null)
        {
            SiteContextOptions.ConfigureHostBuilderAction.Invoke(builder);
        }
        return base.CreateHost(builder);
    }
```